README.md

```diff
@@ -250,6 +250,23 @@ Given a list of file hashes, returns those that are missing in the current root.
 }
 ```
 
+## POST /file/<hash>
+Upload a file with the given hash. Returns a 400 if the hash does not match the file contents.
+This method is idempotent; if the file already exists it will not do anything.
+
+### Body
+The file contents should be written directly to the request body.
+
+### Response
+Returns the location of the file on the server.
+```
+{
+  "status": "success",
+  "errors": null,
+  "data": "/outpack/.outpack/files/sha256/b1/89579a9326f585d308304bd9e03326be5d395ac71b31df359ab8bac408d248"
```

Comment:
why do we return this? when would a consumer need to know where the path was? I would have thought that null would be an ok data response here, along with the success status

Reply:
sure, I guess you don't need it, it's just the standard convention for a POST request to return the location of the created resource (although come to think of it that would be the url `http://packit-instance/file/<hash>`)

Reply:
I think that returning a url would be fine, but I feel reasonably strongly we should not return a path that can't be read from

Reply:
Yeah, ok. Returning null and using the url to test the file has been uploaded.